src/user/linux.rs

```diff
@@ -0,0 +1,102 @@
+use crate::setting::Shell;
+
+use std::fs::File;
+use std::io::{BufRead, BufReader};
+use std::{env, error, fmt};
+
+#[derive(Debug)]
+pub enum Error {
+    SudoUserNotFound,
+    InvalidPasswd,
+    InvalidUserId,
+    InvalidGroupId,
+}
+
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            Error::SudoUserNotFound => write!(f, "user: sudo user not found"),
+            Error::InvalidPasswd => write!(f, "user: invalid passwd"),
+            Error::InvalidUserId => write!(f, "user: invalid userid"),
+            Error::InvalidGroupId => write!(f, "user: invalid groupid"),
+        }
+    }
+}
+
+impl error::Error for Error {}
+
+/// `/etc/passwd`の1行を表す
+struct Passwd {
+    user_name: String,
+    password: String,
+    user_id: u64,
+    group_id: u64,
+    comment: String,
+    home_dir: String,
+    login_shell: Option<Shell>,
+}
+
+impl Passwd {
+    /// `/etc/passwd`の1行をパースする
+    fn parse_passwd_line(line: &str) -> Result<Passwd, Box<dyn std::error::Error>> {
+        let passwd_content: Vec<&str> = line.split(':').collect();
+        let login_shell =
+            if passwd_content[6] == "/usr/sbin/nologin" || passwd_content[6] == "/usr/bin/false" {
+                None
+            } else {
+                match passwd_content[6] {
+                    "/bin/bash" => Some(Shell::Bash),
+                    "/bin/sh" => Some(Shell::Sh),
+                    _ => None,
+                }
+            };
```

Comment:
`nologin`とか`false`のとき特有の処理はしておらず`None`を返すので、matchのdefaultでまとめられそう
```suggestion
        let login_shell = match passwd_content[6] {
                    "/bin/bash" => Some(Shell::Bash),
                    "/bin/sh" => Some(Shell::Sh),
                    _ => None,
                };
```

Reply:
その通りだね。これは恥ずかしいやつ